crates/qasm2/src/parse.rs

```diff
@@ -1685,12 +1696,14 @@ impl State {
             }
             match self.symbols.get(&name) {
                 None => {
+                    // The gate wasn't a built-in, so we need to move the symbol in, but we don't
+                    // need to increment the number of gates because it's already got a gate ID
+                    // assigned.
                     self.symbols.insert(name, symbol.into());
-                    self.num_gates += 1;
-                    Ok(true)
+                    Ok(false)
```

Comment:
This is probably me missing something, because it looks like the tests pass, but if the gate isn't a built-in, shouldn't the function return `Ok(True)` because it needs a definition?

Reply:
It's just off the top of the diff, but this is in an `if` block for a gate that's already in the "overrideable gates" symbol table, which requires the user to have given a constructor function to the Python-space entry point in `custom_instructions`.  So it's already got a gate index allocated, and Python space knows how to construct it.

A "built in" here is meaning "pretend that the gate was always defined in OpenQASM 2", whereas "not a built in" means "use this function to construct the gate, but only allow it if you see a `gate` statement with its name (but ignore the body of that statement)".  We're in the second case here - we don't need to return the body of the gate statement to Python space, because we're ignoring it; we just want to assign the gate index of the overrideable gate to the now properly defined symbol.

Reply:
Thanks! That makes sense, and is what I suspected, but wanted to make sure I was understanding it right.